addrs.go

```diff
@@ -21,22 +53,36 @@ func (addr *Addr) Network() string {
 	return "websocket"
 }
 
-// NewAddr creates a new Addr using the given host string
+// NewAddr creates an Addr with `ws` scheme (insecure).
+//
+// Deprecated. Use NewAddrWithScheme.
 func NewAddr(host string) *Addr {
+	// Older versions of the transport only supported insecure connections (i.e.
+	// WS instead of WSS). Assume that is the case here.
+	return NewAddrWithScheme(host, false)
+}
+
+// NewAddrWithScheme creates a new Addr using the given host string. isSecure
+// should be true for WSS connections and false for WS.
+func NewAddrWithScheme(host string, isSecure bool) *Addr {
```

Comment:
Do we really need an `Addr` constructor? It seems quite trivial to pass in a `url.URL`.